### ISSUE TITLE
Show minimal info page whenever web-interface isn't running?

### ISSUE THREAD

**Issue by JoKeyser** [freedombox/Plinth#915]
This is a question or feature request: When the web-interface fails or gets updated, the user is left without any information, which is quite unsettling if you're not prepared to use SSH. Maybe we could launch a minimal web-server in such cases, to at least give *some* information.
The web-interface routinely isn't available during upgrades of apache2, which can take several minutes if grouped with other packages. Probably it's similar with Django.

Solution sketch: Whenever apache2 isn't running (for whatever reason) launch another, minimal web-server. Preferably its dependencies should not overlap with Plinth's, e.g. no Django, not even Python. Whenever apache2 launches, this goes down.

Maybe this idea can be extended to all critical packages/programs necessary for the web-interface. Maybe we can also group them together during upgrades, to minimize down-time (although I guess that's hard to do, because of dependency clashes?).

**Comment by mray** [comment]
I like this approach to fight FreedomBox going "dark" during updates. *What* kind of information would you expect to show up on this page? I'd already be happy with a simple life sign that makes clear Freedombox is up and running but doing its thing.

**Comment by JoKeyser** [comment]
I agree that a "simple life sign" would go a long way. If possible, it could add a few infos like whether an update is running (which might explain apache2's absence), or the last 10-20 log entries of a few critical services, like Plinth and apache2. Maybe it's actually fine if it was a simple (Python?) process just serving some static HTML; if Python stops running, the trouble is probably beyond (web-based) repair anyhow.
One possible problem is how to keep serving the same (self-signed) server certificate; or maybe we would relay to plain HTTP somehow?

**Comment by mray** [comment]
Not sure about presenting log files to everybody. But hinting about an eventual update and link to a resource that can contain help when things get stuck might suffice?

**Comment by JoKeyser** [comment]
Another note: Maybe this can be done nicely with systemd socket activation, if this allows to specify a primary service (apache2) and some fallback in case it doesn't start?
